### PR TITLE
Feat/37 single blog page part 1

### PR DESCRIPTION
**Pull request by huchshm**
https://website-git-feat-37-single-blog-page-part-1-taodemy.vercel.app/blogs/1

Descriptions
Link to UI design (if possible): [TaoTech](https://www.figma.com/file/q2ZHM8s0LcNxz3otGFLDzO/Website?node-id=0%3A1&t=K8s37vni7t7JiptM-0)
Provide screenshots or videos that the function is working.
working
test

- [x] All unit tests passing and coverage of testing is more than 90% unless special needs.
- [x] No merge conflict with the `main` branch, make sure your code is up to date with `main`.
- [x] Check PR title is aligned with your ticket.
- [x] Check your app will run without crashes.
- [x] Check if there are hard-coded values.
- [x] No `//eslint-disable-next-line no-unused-vars`, unless otherwise specified.
- [x] Check you are using `camelCase && PascalCase` correctly.
- [x] Remove unused comments.

### CODE REVIEW

File: constants/blogDb.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
+{
+  "name": "blogs",
+  "data": [
+    {
+      "id": "1",
+      "title": "Things to Look for When Comparing Branding Alternatives",
+      "subtitle": "lorem Branding is the process of creating a distinct identity for a business in the mind.",
```
Comment (Cameron933) [inline constants/blogDb.json:7]: Make it meaningful

File: pages/blogs/[id].tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  useEffect(() => {
+    const blog = blogs.data.filter((blog) => blog.id === id);
+    if (blog[0]) setCurrentBlog(blog[0]);
+  }, [id]);
+
+  return currentBlog ? (
+    <div>
+      <section className={`${styles.blog_title_section}`}>
+        <div className={`${styles.blog_title_button}`}>
+          <Button variant="secondary" padding="small">
```
Comment (Cameron933) [inline pages/blogs/[id].tsx:35]: This button here should be able to take the user back to the blog page

File: constants/blogDb.json [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      "content": "Branding is the process of creating a distinct creating a distinct identity for a business in the mind. loremcreating a distinct identity for a business in the mind. loremcreating a distinct identity for a business in the mind. loremcreating a distinct identity for a business in the mind. loremcreating a distinct identity for a business in the mind. loremcreating a distinct identity for a business in the mind. loremidentity for a business in the mind. lorem",
+      "image": "/images/two_people_praying.png",
+      "alt": "Branding icon",
+      "info": {
+        "date": "16 JUNE 2022",
+        "category": "design",
+        "readingTime": "10 MIN"
+      },
+      "author": "Guy Hawkins",
+      "authorInfo": "lorem Branding is the process of creating a distinct "
```
Comment (Cameron933) [inline constants/blogDb.json:17]: Make it meaningful instead of meaningless lorem

File: pages/blogs/[id].tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  const router = useRouter();
+  const { id } = router.query;
+  console.log(typeof id);
+  useEffect(() => {
+    const blog = blogs.data.filter((blog) => blog.id === id);
+    if (blog[0]) setCurrentBlog(blog[0]);
+  }, [id]);
+
+  return currentBlog ? (
+    <div>
```
Comment (Cameron933) [inline pages/blogs/[id].tsx:32]: why a div outside of the section? you can just use <> </> as parent root

File: styles/BlogSingle.module.css [hunk 5]
```diff
@@ -0,0 +1,62 @@
+.blog_image_container {
+  width: 1780px;
```
Comment (Cameron933) [inline styles/BlogSingle.module.css:2]: We should not define a fixed width, maybe you can think about using a grid here.

File: pages/blogs/[id].tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+  author: string;
+  authorInfo: string;
+}
+export default function Blog() {
+  const [currentBlog, setCurrentBlog] = useState<Blog | null>(null);
+  const router = useRouter();
+  const { id } = router.query;
+  console.log(typeof id);
+  useEffect(() => {
+    const blog = blogs.data.filter((blog) => blog.id === id);
```
Comment (tatoMa) [inline pages/blogs/[id].tsx:27]: Have a look at JS array find method. The filter is the one works but not the best solution.

**Review by tatoMa — COMMENTED** [review]
Think about restructuring and extracting some of your repeated and relatively complicated parts into another file or part.

File: pages/blogs/[id].tsx [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+  const [currentBlog, setCurrentBlog] = useState<Blog | null>(null);
+  const router = useRouter();
+  const { id } = router.query;
+  useEffect(() => {
+    const blog = blogs.data.find((blog) => blog.id === id);
+    if (blog) setCurrentBlog(blog);
+  }, [id]);
+
+  return currentBlog ? (
+    <>
```
Comment (jiayuu-u) [inline pages/blogs/[id].tsx:32]: Why do we still need react fragment here?

File: pages/blogs/[id].tsx [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+              <img src="/images/badge-fb.svg" alt="badge" aria-label="badge__fb" />
+            </li>
+            <li>
+              <img src="/images/badge-in.svg" alt="badge" aria-label="badge__in" />
+            </li>
+            <li>
+              <img src="/images/badge-ln.svg" alt="badge" aria-label="badge__ln" />
+            </li>
+          </ul>
+        </section>
```
[the remark this answers is not in the document]
Comment (huchshm) [reply pages/blogs/[id].tsx:82]: 已改

File: constants/blogDb.json [hunk 9]
```diff
@@ -0,0 +1,65 @@
+{
+  "name": "blogs",
+  "data": [
+    {
+      "id": "1",
+      "title": "Things to Look for When Comparing Branding Alternatives",
```
[the remark this answers is not in the document]
Comment (huchshm) [reply constants/blogDb.json:6]: fixted

File: constants/blogDb.json [hunk 10]
```diff
@@ -0,0 +1,65 @@
+{
+  "name": "blogs",
+  "data": [
+    {
+      "id": "1",
+      "title": "Things to Look for When Comparing Branding Alternatives",
+      "subtitle": "lorem Branding is the process of creating a distinct identity for a business in the mind.",
+      "content": "Branding is the process of creating a distinct creating a distinct identity for a business in the mind. loremcreating a distinct identity for a business in the mind. loremcreating a distinct identity for a business in the mind. loremcreating a distinct identity for a business in the mind. loremcreating a distinct identity for a business in the mind. loremcreating a distinct identity for a business in the mind. loremidentity for a business in the mind. lorem",
```
[the remark this answers is not in the document]
Comment (huchshm) [reply constants/blogDb.json:8]: fitxed

File: constants/blogDb.json (hunk 1 above)
Comment (huchshm) [reply constants/blogDb.json:7]: fixed

File: constants/blogDb.json (hunk 9 above)
[the remark this answers is not in the document]
Comment (huchshm) [reply constants/blogDb.json:6]: image
fixed.

**Review by HarryuCATGREY — APPROVED** [review]
LGTM

**Review by Trent1900 — APPROVED** [review]
LGTM

**Pull request merged**